Bry/styling

bry-gavino opened the pull request:
# ✨ New in this PR ✨

## One liner: what did you do? 🚨 

## Coverage 🙆‍♀️
_use this section to break up your task into submodules and track progress_ 
 - [X] keyboard doesn't cover up the text field
 - [X] removed notifications button -- put Log Out button in settings tab
 - [X] added clickable redirection links to device's email/phone-call app
 -  [X] fixed upload screen to scroll
 - [X] _**(EDITED)**_: update button does NOT work -- this is being handled in **bry/error-alerts** branch  _(OUTDATED) fixed update button to work_


## How can the reviewer test your code? 👩‍💻
 - keyboard doesn't cover up the text field
      - you should be able to see what you're typing in: Log in, Register, Settings, Intake Form screens
 
 - removed notifications button -- put Log Out button in settings tab
      - tap on the settings button
      - Logout button should log you out
      - update button should be placed underneath the Radio content
      
 - added clickable redirection links to device's email/phone-call app
      - at the top of the Home screen
![IMG_01897643F8E0-1](https://user-images.githubusercontent.com/60284767/169627442-1cf091f8-8e72-4d6e-8320-e3b9383d8040.jpeg)
      
 -  fixed upload screen to scroll
      - have multiple cases filled out
      - check that you can scroll and view all the Doc Containers for each case
      
 - fixed update button to work
      - in the code, refractored auth.ts functions updateEmail() and updateFirebaseLanguage() to take in clientId
<img width="585" alt="Screen Shot 2022-05-20 at 5 40 09 PM" src="https://user-images.githubusercontent.com/60284767/169627416-e461ac76-7678-4656-8a02-7a906ff70d4d.png">


## Updates
- **Fixed Register Problem**: 
    - **Problem**: Registering an account would register an account in the Firebase authentication but not create a client in the database. Thus, the Home Screen would not display the user's name.
    - **Fixed Functionality**: Registering an account will now create a client in the database and the Home Screen now displays the user's name.
    - **How it was fixed:** database/auth.ts register() now uses usr.user instead of user 
<img width="409" alt="Screen Shot 2022-05-27 at 3 13 01 PM" src="https://user-images.githubusercontent.com/60284767/170795421-3a04f347-cca1-4f02-bde4-4d825ff88c9e.png">



🧜‍♀️ cc: @shannonbonet

shannonbonet commented on src/screens/settings/SettingsScreen.tsx:
```diff
@@ -25,6 +30,67 @@ const SettingsScreen = ({ navigation }: any) => {
+  const { userLanguageChange } = React.useContext(LanguageContext);
+  const { state } = React.useContext(ClientContext);
+
+  const updateLanguage = async (lang: string) => {
```
I moved these functions to database/auth to separate code we render from backend queries so could you apply this refactor to those pre-existing functions?

bry-gavino replied:
Ty--and yup, I'll refractor the pre-existing functions.

shannonbonet commented on src/screens/settings/SettingsScreen.tsx:
```diff
@@ -49,21 +116,39 @@ const SettingsScreen = ({ navigation }: any) => {
-      <TextSubtitle>{Text('Go Back')}</TextSubtitle>
-    </ButtonHeader>
-  );
+  const handleRadio = (val: string): void => {
```
I moved this code to a helper function called `updateLanguage` in`utils` which handles a string and dictionary update. Read the difference here: https://github.com/calblueprint/siren/pull/79#issue-1242509140

shannonbonet replied:
```
export const updateLanguage = (
  val: string,
  dictUpdate: any, // JSON DICTIONARY UPDATE FUNCTION  
  stringUpdate: any, // FIREBASE LANGUAGE FIELD UPDATE FUNCTION 
) => {
  if (val === 'Español') {
    dictUpdate(dictionaryList.ES); // DICT TYPE
    stringUpdate('ES'); // STRING TYPE
  }
  if (val === 'Tiếng Việt') {
    dictUpdate(dictionaryList.VIET);
    stringUpdate('VIET');
  }
  if (val === 'English') {
    dictUpdate(dictionaryList.EN);
    stringUpdate('EN');
  }
};```

shannonbonet commented on src/screens/settings/SettingsScreen.tsx:
```diff
@@ -25,6 +30,67 @@ const SettingsScreen = ({ navigation }: any) => {
   const [currentPassword, setCurrentPassword] = useState('');
+  const db = firebase.firestore();
+  const clientCollection = db.collection('clients');
+  const { userLanguageChange } = React.useContext(LanguageContext);
```
This won't work since LanguageContext stores a dictionary type, and you are updating it to a string type in the handleRadio function below. Refer to explanation [here. ](https://github.com/calblueprint/siren/pull/79)

> Changing language must reflect in 2 places:

> language context sets the json language dictionary (ES, EN, VIET)
> setting client's language in Firebase.
> Context Providers are used for persisting user states in the app. They are also useful for reflecting changes that the user needs to see immediately. This means LanguageProvider only has to save the state of current language dictionary, since all static texts we see across the app use a local dictionary's translations.
> 
> For dynamic texts (texts that SIREN will change), we defer to Firebase storage. Intake form questions are the only texts subject to change. This means we store a language field for the client, which we use to query questions with based on current client's language. We save this as a string ES/VIET/EN. Though we could've saved this as apart of the Language Provider, there isn't a large need for this since intake form pulls from Firebase, not the local app language.

shannonbonet reviewed (commented):
In the middle of review

So far, `SettingScreen` should be updated to reflect the multilingual refactor. The main distinction is that we are using a string type to update firebase, and a dictionary state (stored as `LanguageContext`) for all other static texts

shannonbonet commented:
**Couple of things:** 

- put the update + logout button in the same container and center content in the settings page
- when clicking update, route the user back to the home page
- fix padding of the home page title to be in line with rest of content
<img width="364" alt="Screen Shot 2022-06-01 at 4 37 38 PM" src="https://user-images.githubusercontent.com/63089231/171497115-83e20302-dc8e-4852-a42d-450e0cc2c520.png">

- move contact buttons below the progress tracker @ bottom of the page to match designs. Please use the styled components in` assets/Components.tsx`. Look [here](https://github.com/calblueprint/siren/blob/db0f6f44b956bcfc07846a3a577f0d2bb2f68fce/src/screens/Welcome/WelcomeScreen.tsx#L51) for example of ButtonLight usage. You should use this component too. 
- 
<img width="259" alt="Screen Shot 2022-06-01 at 4 43 59 PM" src="https://user-images.githubusercontent.com/63089231/171498087-a00212df-082e-436d-9538-96a784383c11.png">

- fix layout of the Register heading page - the title is overlapping with the go back button 
<img width="348" alt="Screen Shot 2022-06-01 at 4 36 15 PM" src="https://user-images.githubusercontent.com/63089231/171496887-4d8999ee-7bab-4687-a243-9c7d6528e3a6.png">

- Could you clean up the Settings screen to use refactored in `database/auth` queries? There is a bunch of dead + duplicate code that needs clean up before merging. Please refer to the multilingual refactor (or current version of main) for where these functions live and adapt any refactoring you've done appropriately. This is the primary reason I've not merged yet.

bry-gavino commented:
> **Couple of things:**
> 
> * put the update + logout button in the same container and center content in the settings page
> * when clicking update, route the user back to the home page
> * fix padding of the home page title to be in line with rest of content
> 
> <img alt="Screen Shot 2022-06-01 at 4 37 38 PM" width="364" src="https://user-images.githubusercontent.com/63089231/171497115-83e20302-dc8e-4852-a42d-450e0cc2c520.png">
> 
> * move contact buttons below the progress tracker @ bottom of the page to match designs. Please use the styled components in` assets/Components.tsx`. Look [here](https://github.com/calblueprint/siren/blob/db0f6f44b956bcfc07846a3a577f0d2bb2f68fce/src/screens/Welcome/WelcomeScreen.tsx#L51) for example of ButtonLight usage. You should use this component too.
> 
> <img alt="Screen Shot 2022-06-01 at 4 43 59 PM" width="259" src="https://user-images.githubusercontent.com/63089231/171498087-a00212df-082e-436d-9538-96a784383c11.png">
> 
> * fix layout of the Register heading page - the title is overlapping with the go back button
> 
> <img alt="Screen Shot 2022-06-01 at 4 36 15 PM" width="348" src="https://user-images.githubusercontent.com/63089231/171496887-4d8999ee-7bab-4687-a243-9c7d6528e3a6.png">
> 
> * Could you clean up the Settings screen to use refactored in `database/auth` queries? There is a bunch of dead + duplicate code that needs clean up before merging. Please refer to the multilingual refactor (or current version of main) for where these functions live and adapt any refactoring you've done appropriately. This is the primary reason I've not merged yet.



_Addressed the issues listed in this comment -- ty @shannonbonet ! _

**One foreseeable issue + blocker**:
- In my iPhone/Android simulators, the Go Back button did NOT overlap with the Register header in the Register Screen but did so for Shannon's end. I am unsure of how to approach this such that all smartphones will not have styling issues. 
  _- I have tried to use the width of the Go Back button as a margin-top value for Register Header, but it appears that the content of the Register Header + input fields and Go Back button do not have the same "points of reference" in terms of their placements on the screen_
